### DISCUSSION TITLE
How do I do an "early return"?

### DISCUSSION THREAD

**Discussion by mrtry** [google/zx#657] [Help needed]
I wanted to check the flag and return the shell script early.
I tried two patterns, but they did not seem to work.

In the following case, `exit 0` was evaluated, but code execution did not stop and subsequent code blocks were also evaluated.

```js
if ($.env.APP_VARIANT === 'debug') {
  await $`exit 0`
}
```

The following will result in an error `SyntaxError: Illegal return statement`.

```js
if ($.env.APP_VARIANT === 'debug') {
  return await $`exit 0``
}
```

This also results in a syntax error.

```js
if ($.env.APP_VARIANT === 'debug') {
  return;
}
```

The version of zx is "7.2.3".

What should I do?

**Comment by antongolub** [comment]
You can wrap the entire code into IIFE, so `return` would be legal syntax.

**Comment by modjke** [comment]
`process.exit(0)` is a cleaner alternative with an option to specify exit code
